File: refrigitz 12/NetworkQuantumLearningKrinskyAtamata.cpp
#include "stdafx.h"


namespace RefrigtzDLL
{

	std::wstring NetworkQuantumLearningKrinskyAtamata::Root = L""; //System::IO::Path::GetDirectoryName(Environment::GetCommandLineArgs());

	/*void NetworkQuantumLearningKrinskyAtamata::Log(std::exception ex)
	{
		//try
		{
			//autoa = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
			//lock (a)
			{
				std::wstring stackTrace = ex.what();
//C# TO C++ CONVERTER TODO TASK: There is no native C++ equivalent to 'ToString':
				File::AppendAllText(Root + std::wstring(L"\\ErrorProgramRun.txt"), stackTrace + std::wstring(L": On") + DateTime::Now.ToString()); // path of file where stack trace will be stored.
			}
		}
		//catch(std::exception t)
		{
			
		}
	}
	*/
	NetworkQuantumLearningKrinskyAtamata::NetworkQuantumLearningKrinskyAtamata(int r0, int m0, int k0) : LearningKrinskyAtamata(r0, m0, k0)
	{
		InitializeInstanceFields();
		//autoo = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
		//lock (o)
		{
			Netfi = new LearningKrinskyAtamata*[k0];
			for (int i = 0; i < m0; i++)
				Netfi[i] =new LearningKrinskyAtamata[m0];

			for (int j = 0; j < m0; j++)
			{
				for (int k = 0; k < k0; k++)
				{
					Netfi[j][k] = new LearningKrinskyAtamata(r0, m0, k0);
				}
			}

			r = r0;
			m = m0;
			k = k0;
		}
	}

	double NetworkQuantumLearningKrinskyAtamata::LearningAlgorithmRegard(int Row, int Column)
	{
		//autoo = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
		//lock (o)
		{

			double Hu = 1;
			//try
			{
				Netfi[Row][Column].LearningAlgorithmRegard();
				Hu = Netfi[Row][Column].alpha[State];
			}
			//catch(std::exception t)
			{
				
			}
			return Hu;
		}
	}

	

	double NetworkQuantumLearningKrinskyAtamata::IsRewardAction(int Row, int Column)
	{
		//autoo = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
		//lock (o)
		{
			if (Netfi[Row][Column].IsReward)
			{
				return 1;
			}
			return -1;
		}
	}

	double NetworkQuantumLearningKrinskyAtamata::IsPenaltyAction(int Row, int Column)
	{
		//autoo = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
		//lock (o)
		{
			if (Netfi[Row][Column].IsPenalty)
			{
				return 0;
			}
			return -1;
		}
	}

	double NetworkQuantumLearningKrinskyAtamata::LearningAlgorithmPenalty(int Row, int Column)
	{
		//autoo = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
		//lock (o)
		{
			double Hu = 1;
			//try
			{
				Netfi[Row][Column].LearningAlgorithmPenalty();
				Hu = Netfi[Row][Column].alpha[State];
			}
			//catch(std::exception t)
			{
				
			}
			return Hu;
		}
	}

	double NetworkQuantumLearningKrinskyAtamata::LearingValue(int Row, int Column) 
	{
			double Hu = 1;
			//try
			{
				Hu = Netfi[Row][Column].alpha[State];
			}
			//catch(std::exception t)
			{
				
			}
			return Hu;
		
	}

	void NetworkQuantumLearningKrinskyAtamata::InitializeInstanceFields()
	{
		r = 0;
		m = 0;
		k = 0;
	}
}

File: ChessFirst/NetworkQuantumLearningKrinskyAtamata.cs
/*CopyRight Ramin Edjlal***************************2018*************************
 The Magic Table Game Satte Learing Quantum Atamata.****************************
 *******************************************************************************
 */
using System;
using System.IO;

namespace ChessFirst
{
    [Serializable]
    public class NetworkQuantumLearningKrinskyAtamata : LearningKrinskyAtamata
    {
        public static String Root = System.IO.Path.GetDirectoryName(Environment.GetCommandLineArgs()[0]);
        static void Log(Exception ex)
        {

            Object a = new Object();
            lock (a)
            {
                string stackTrace = ex.ToString();
                File.AppendAllText(Root + "\\ErrorProgramRun.txt", stackTrace + ": On" + DateTime.Now.ToString()); // path of file where stack trace will be stored.
            }

        }

        int r, m, k;
        LearningKrinskyAtamata[,] Netfi;


        public NetworkQuantumLearningKrinskyAtamata(int r0, int m0, int k0) : base(r0, m0, k0)
        {
            Object o = new Object();
            lock (o)
            {
                Netfi = new LearningKrinskyAtamata[m0, k0];

                for (int j = 0; j < m0; j++)
                    for (int p = 0; p < k0; p++)
                        Netfi[j, p] = new LearningKrinskyAtamata(r0, m0, k0);

                r = r0;
                m = m0;
                k = k0;
            }
        }
        public double LearningAlgorithmRegardNet(int Row, int Column)
        {
            Object o = new Object();
            lock (o)
            {

                double Hu = 1;

                Netfi[Row, Column].LearningAlgorithmRegard();
                Hu = Netfi[Row, Column].alpha[State];

                return Hu;
            }
        }
        public int IsRewardActionNet(int Row, int Column)
        {
            Object o = new Object();
            lock (o)
            {
                if (Netfi[Row, Column].IsReward)
                    return 1;
                return -1;
            }
        }

        public double IsPenaltyActionNet(int Row, int Column)
        {
            Object o = new Object();
            lock (o)
            {
                if (Netfi[Row, Column].IsPenalty)
                    return 0;
                return -1;
            }
        }
        public double LearningAlgorithmPenaltyNet(int Row, int Column)
        {
            Object o = new Object();
            lock (o)
            {
                double Hu = 1;

                Netfi[Row, Column].LearningAlgorithmPenalty();
                Hu = Netfi[Row, Column].alpha[State];

                return Hu;
            }
        }
        public double LearingValue(int Row, int Column)
        {
            Object o = new Object();
            lock (o)
            {
                double Hu = 1;

                Hu = Netfi[Row, Column].alpha[State];

                return Hu;
            }
        }
    }
}

